app/src/main/java/org/mozilla/fenix/settings/search/SearchEngineListPreference.kt

```diff
@@ -215,12 +215,9 @@ abstract class SearchEngineListPreference @JvmOverloads constructor(
             },
             operation = {
                 if (isDefaultEngine) {
-                    context.settings().defaultSearchEngineName = context
-                        .components
-                        .search
-                        .provider
-                        .getDefaultEngine(context)
-                        .name
+                    val default = context.components.search.provider.getDefaultEngine(context)
+                    context.components.search.provider.setDefaultEngine(context, default.identifier)
```

Comment:
Do you think the `context.settings().defaultSearchEngineName = default.name` should be moved into `setDefaultEngine`? So that we know it will always be set correctly?